src/ir/comp.rs

```diff
@@ -490,49 +490,67 @@ fn bitfields_to_allocation_units<E, I>(ctx: &BindgenContext,
     let mut unit_align = 0;
     let mut bitfields_in_unit = vec![];
 
+    // TODO(emilio): Determine this from attributes or pragma ms_struct
+    // directives. Also, perhaps we should check if the target is MSVC?
+    const is_ms_struct: bool = false;
+
     for bitfield in raw_bitfields {
         let bitfield_width = bitfield.bitfield().unwrap() as usize;
-        let bitfield_align = ctx.resolve_type(bitfield.ty())
-            .layout(ctx)
-            .expect("Bitfield without layout? Gah!")
-            .align;
-
-        if unit_size_in_bits != 0 &&
-           (bitfield_width == 0 ||
-            bitfield_width > unfilled_bits_in_unit) {
-            // We've reached the end of this allocation unit, so flush it
-            // and its bitfields.
-            unit_size_in_bits = align_to(unit_size_in_bits,
-                                                 bitfield_align);
-            flush_allocation_unit(fields,
-                                  bitfield_unit_count,
-                                  unit_size_in_bits,
-                                  unit_align,
-                                  mem::replace(&mut bitfields_in_unit, vec![]));
-
-            // Now we're working on a fresh bitfield allocation unit, so reset
-            // the current unit size and alignment.
-            unit_size_in_bits = 0;
-            unit_align = 0;
+        let bitfield_layout =
+            ctx.resolve_type(bitfield.ty())
+                .layout(ctx)
+                .expect("Bitfield without layout? Gah!");
+        let bitfield_size = bitfield_layout.size;
+        let bitfield_align = bitfield_layout.align;
+
+        let mut offset = unit_size_in_bits;
+        if is_ms_struct {
+            if unit_size_in_bits != 0 &&
+               (bitfield_width == 0 ||
+                bitfield_width > unfilled_bits_in_unit) {
+                // We've reached the end of this allocation unit, so flush it
+                // and its bitfields.
+                unit_size_in_bits = align_to(unit_size_in_bits, unit_align * 8);
+                flush_allocation_unit(fields,
+                                      bitfield_unit_count,
+                                      unit_size_in_bits,
+                                      unit_align,
+                                      mem::replace(&mut bitfields_in_unit, vec![]));
+
+                // Now we're working on a fresh bitfield allocation unit, so reset
+                // the current unit size and alignment.
+                #[allow(unused_assignments)]
```

Comment:
Why are these assignments considered unused? Because we are always `is_ms_struct == false`?